apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs

```diff
@@ -260,8 +271,17 @@ private SpannerCommand GetBuiltDeleteCommand()
         {
             if (_builtDeleteCommand == null && _parsedParameterCollection != null && AutoCreateCommands)
             {
+                SpannerParameterCollection deleteCollection = new SpannerParameterCollection();
+                //Filter the delete collection down to just the ones representing the primary keys.
+                foreach(SpannerParameter parameter in _parsedParameterCollection)
+                {
+                    if (AutoGeneratedCommandPrimaryKeys.Contains(parameter.ParameterName))
```

Comment:
often (but surely not always) a parameter indicates a column.

eg in an insertcommand, the parameters are column values and each parameter corresponds to a column.  In fact, each parameter in ADO.NET has a source column name from which to pull the value from a datatable (which is usually either originally a column from the database or an expression from a query).

So its not weird.  Another way of thinking about it is that parameters are a superset that comprises of column values but can be other things (like inputs to a where clause or sproc).  

For some spanner commands (due to not supporting DDL), parameters can only indicate columns (insert, delete, update). This was all in my original design doc.

In this particular case, the commands for insert, delete and update are being built automatically by the dataadapter and therefore the parameter set up per column is done for the developer.